src/compiler/checker.ts

```diff
@@ -29471,14 +29471,36 @@ namespace ts {
                 for (let i = spreadIndex; i < args.length; i++) {
                     const arg = args[i];
                     // We can call checkExpressionCached because spread expressions never have a contextual type.
-                    const spreadType = arg.kind === SyntaxKind.SpreadElement && (flowLoopCount ? checkExpression((arg as SpreadElement).expression) : checkExpressionCached((arg as SpreadElement).expression));
-                    if (spreadType && isTupleType(spreadType)) {
-                        forEach(getTypeArguments(spreadType), (t, i) => {
-                            const flags = spreadType.target.elementFlags[i];
-                            const syntheticArg = createSyntheticExpression(arg, flags & ElementFlags.Rest ? createArrayType(t) : t,
-                                !!(flags & ElementFlags.Variable), spreadType.target.labeledElementDeclarations?.[i]);
-                            effectiveArgs.push(syntheticArg);
-                        });
+                    const argType = arg.kind === SyntaxKind.SpreadElement && (flowLoopCount ? checkExpression((arg as SpreadElement).expression) : checkExpressionCached((arg as SpreadElement).expression));
+                    if (argType) {
+                        let spreadType: TupleTypeReference | undefined;
+                        if (!!(argType.flags & TypeFlags.Intersection)) {
+                            for (const type of (argType as IntersectionType).types) {
+                                if (isTupleType(type)) {
+                                    if (spreadType) {
+                                        // If there's multiple tuples in this intersection,
+                                        // do not try to resolve into multiple arguments.
+                                        spreadType = undefined;
+                                        break;
+                                    }
+                                    spreadType = type;
+                                }
+                            }
+                        }
+                        else {
+                            spreadType = isTupleType(argType) ? argType : undefined;
+                        }
+                        if (spreadType) {
+                            forEach(getTypeArguments(spreadType), (t, i) => {
+                                const flags = spreadType!.target.elementFlags[i];

```

Comment:
Is the new `!` still needed? It looks like it shouldn't be, but I can't be sure.

Reply:
It is still required as TypeScript reverts to the original variable type of (TupeTypeReference | undefined) inside of functions.